nmt: prune's idxStart, idxEnd, maxWidth bounds aren't checked

This code in prune doesn't have sanity bounds checks and these vectors cause panics
```go
prune(0, 0, 0)
prune(0, 1, 0)
prune(0, 1, 1)
prune(1, 0, 0)
```

```shell
$ go test -run=Prune -fuzz=^$ -short
--- FAIL: TestPrune (0.00s)
panic: runtime error: slice bounds out of range [:-1] [recovered]
	panic: runtime error: slice bounds out of range [:-1]

goroutine 18 [running]:
testing.tRunner.func1.2({0x12271e0, 0xc00012c018})
	/Users/emmanuelodeke/go/src/go.googlesource.com/go/src/testing/testing.go:1451 +0x24e
testing.tRunner.func1()
	/Users/emmanuelodeke/go/src/go.googlesource.com/go/src/testing/testing.go:1454 +0x39f
panic({0x12271e0, 0xc00012c018})
	/Users/emmanuelodeke/go/src/go.googlesource.com/go/src/runtime/panic.go:884 +0x213
github.com/celestiaorg/nmt.prune(0x0, 0x0, 0x0)
	/Users/emmanuelodeke/go/src/github.com/celestiaorg/nmt/subrootpaths.go:52 +0x6e5
github.com/celestiaorg/nmt.TestPrune(0x0?)
	/Users/emmanuelodeke/go/src/github.com/celestiaorg/nmt/vuln_test.go:10 +0x45
testing.tRunner(0xc000116820, 0x12479f8)
	/Users/emmanuelodeke/go/src/go.googlesource.com/go/src/testing/testing.go:1501 +0x10b
created by testing.(*T).Run
	/Users/emmanuelodeke/go/src/go.googlesource.com/go/src/testing/testing.go:1548 +0x35f
exit status 2
FAIL	github.com/celestiaorg/nmt	0.369s
```

--- a/subrootpaths.go
+++ b/subrootpaths.go
@@ -16,7 +16,7 @@ var (
 // this is just a quick function to return that representation as a list of ints
 func subdivide(idxStart uint, width uint) []int {
 	var path []int
-	pathlen := int(math.Log2(float64(width)))
+	pathlen := int(bits.Len(width) - 1)
 	for i := pathlen - 1; i >= 0; i-- {
 		if (idxStart & (1 << i)) == 0 {
 			path = append(path, 0)
@@ -37,9 +37,12 @@ func extractBranch(path []int, index int, offset int, branch int) []int {
 }
 
 func prune(idxStart uint, idxEnd uint, maxWidth uint) [][]int {
-
-	var prunedPaths [][]int
-	var preprocessedPaths [][]int
+	if idxEnd == 0 || maxWidth == 0 {
+		return nil
+	}
+	if idxStart > idxEnd || idxEnd >= maxWidth {
+		return nil
+	}
 
 	pathStart := subdivide(idxStart, maxWidth)
 	pathEnd := subdivide(idxEnd, maxWidth)
@@ -52,6 +55,9 @@ func prune(idxStart uint, idxEnd uint, maxWidth uint) [][]int {
 			return [][]int{pathStart[:len(pathStart)-1]}
 		}
 	}
+
+	var prunedPaths [][]int
+	var preprocessedPaths [][]int
 
 	// if starting share is on an odd index, add that single path and shift it right 1
 	if idxStart%2 == 1 {

